### ISSUE TITLE
ERROR: Could not resolve via dns

### ISSUE THREAD

**Issue by nmvh5** [nickbabcock/dness#184]
When trying to update IP address to cloudflare, it returns with error "could not successfully resolve IP caused by: could not resolve via dns"

Trace level output is:

C:\Apps\dness>C:\Apps\dness\dness-v0.2.1.exe -c C:\Apps\dness\dness.conf
[DEBUG trust_dns_resolver::async_resolver] trust-dns resolver running
[TRACE trust_dns_resolver::async_resolver] handle passed back
[DEBUG trust_dns_proto::xfer::dns_handle] querying: myip.opendns.com. A
[DEBUG trust_dns_resolver::name_server::name_server_pool] sending request: [Query { name: Name { is_fqdn: true, labels: [myip, opendns, com] }, query_type: A, query_class: IN }]
[DEBUG trust_dns_resolver::name_server::name_server] reconnecting: NameServerConfig { socket_addr: V4(208.67.222.222:53), protocol: Udp, tls_dns_name: None }
[DEBUG trust_dns_proto::xfer] enqueueing message: [Query { name: Name { is_fqdn: true, labels: [myip, opendns, com] }, query_type: A, query_class: IN }]
[TRACE trust_dns_resolver::name_server::connection_provider] polling response inner
[DEBUG trust_dns_resolver::name_server::name_server] reconnecting: NameServerConfig { socket_addr: V4(208.67.220.220:53), protocol: Udp, tls_dns_name: None }
[DEBUG trust_dns_proto::xfer] enqueueing message: [Query { name: Name { is_fqdn: true, labels: [myip, opendns, com] }, query_type: A, query_class: IN }]
[TRACE trust_dns_resolver::name_server::connection_provider] polling response inner
[TRACE trust_dns_resolver::name_server::connection_provider] polling response inner
[TRACE trust_dns_resolver::name_server::connection_provider] polling response inner
[TRACE mio::sys::windows::selector] select; timeout=Some(0ns)
[TRACE mio::sys::windows::selector] polling IOCP
[TRACE mio::sys::windows::selector] returning
[TRACE trust_dns_resolver::name_server::connection_provider] polling response inner
[TRACE mio::poll] registering with poller
[TRACE mio::sys::windows::selector] register Token(0) Readable | Writable
[TRACE mio::sys::windows::selector] set readiness to (empty)
[TRACE mio::sys::windows::udp] scheduling a read
[TRACE mio::sys::windows::selector] set readiness to Writable
[DEBUG trust_dns_proto::udp::udp_stream] created socket successfully
[TRACE trust_dns_resolver::name_server::connection_provider] polling response inner
[TRACE mio::poll] registering with poller
[TRACE mio::sys::windows::selector] register Token(1) Readable | Writable
[TRACE mio::sys::windows::selector] set readiness to (empty)
[TRACE mio::sys::windows::udp] scheduling a read
[TRACE mio::sys::windows::selector] set readiness to Writable
[DEBUG trust_dns_proto::udp::udp_stream] created socket successfully
[TRACE mio::sys::windows::selector] select; timeout=Some(0ns)
[TRACE mio::sys::windows::selector] polling IOCP
[TRACE mio::sys::windows::selector] returning
[TRACE trust_dns_resolver::name_server::connection_provider] polling response inner
[TRACE mio::sys::windows::selector] set readiness to (empty)
[TRACE mio::sys::windows::udp] scheduling a send
[TRACE trust_dns_resolver::name_server::connection_provider] polling response inner
[TRACE mio::sys::windows::selector] set readiness to (empty)
[TRACE mio::sys::windows::udp] scheduling a send
[TRACE mio::sys::windows::selector] select; timeout=Some(0ns)
[TRACE mio::sys::windows::selector] polling IOCP
[TRACE mio::sys::windows::selector] select; -> got overlapped
[TRACE mio::sys::windows::udp] finished a send 45
[TRACE mio::sys::windows::selector] set readiness to Writable
[TRACE mio::sys::windows::selector] select; -> got overlapped
[TRACE mio::sys::windows::udp] finished a send 45
[TRACE mio::sys::windows::selector] set readiness to Writable
[TRACE mio::sys::windows::selector] returning
[TRACE trust_dns_resolver::name_server::connection_provider] polling response inner
[TRACE trust_dns_resolver::name_server::connection_provider] polling response inner
[TRACE mio::sys::windows::selector] select; timeout=Some(0ns)
[TRACE mio::sys::windows::selector] polling IOCP
[TRACE mio::sys::windows::selector] returning
[TRACE trust_dns_resolver::name_server::connection_provider] polling response inner
[TRACE trust_dns_resolver::name_server::connection_provider] polling response inner
[TRACE mio::sys::windows::selector] select; timeout=Some(4.0762395s)
[TRACE mio::sys::windows::selector] polling IOCP
[TRACE mio::sys::windows::selector] select; -> got overlapped
[TRACE mio::sys::windows::udp] finished a recv 91
[TRACE mio::sys::windows::selector] set readiness to Readable | Writable
[TRACE mio::sys::windows::selector] returning
[TRACE trust_dns_resolver::name_server::connection_provider] polling response inner
[TRACE trust_dns_resolver::name_server::connection_provider] polling response inner
[TRACE mio::sys::windows::selector] set readiness to Writable
[TRACE mio::sys::windows::udp] scheduling a read
[DEBUG trust_dns_proto::rr::record_data] reading SOA
[DEBUG trust_dns_proto::udp::udp_client_stream] received message id: 24875
[TRACE mio::poll] deregistering handle with poller
[TRACE mio::sys::windows::selector] deregistering
[TRACE mio::poll] deregistering handle with poller
[TRACE mio::sys::windows::selector] deregistering
[DEBUG trust_dns_resolver::name_server::name_server_pool] mDNS responsed for query: NXDomain
[ERROR dness] could not successfully resolve IP
        caused by: could not resolve via dns

**Comment by nickbabcock** [comment]
Fascinating, I don't believe that this is due to cloudflare. I believe the method used to get your WAN IP is failing. I'm assuming you get the same error when executing dness without a config (like shown)

```
dness-v0.2.1.exe
```

If you don't get an error, let me.

I believe that you will also receive an error if you execute the following:

```
nslookup myip.opendns.com. 208.67.220.220
```
or
```
nslookup myip.opendns.com. 208.67.222.222
```

(again, if no error, let me know).

This may be due to some computer / ISP setting that is blocking opendns or that domain and the potential fix would be to adding the ability to resolve the WAN address through other means than DNS.

**Comment by nmvh5** [comment]
It does appear to be failing trying to get my WAN IP. 


>dness-v0.2.1.exe
[ERROR dness] could not successfully resolve IP
        caused by: could not resolve via dns

-------------------------------------------------------------

>nslookup myip.opendns.com. 208.67.220.220
Server:  resolver2.opendns.com
Address:  208.67.220.220

-------------------------------------------------------------
>nslookup myip.opendns.com. 208.67.220.220
Server:  resolver2.opendns.com
Address:  208.67.220.220

*** resolver2.opendns.com can't find myip.opendns.com.: Non-existent domain

*** resolver2.opendns.com can't find myip.opendns.com.: Non-existent domain

**Comment by nickbabcock** [comment]
Ok thank you for your quick response. I wonder if your network will allow alternative ways to get your IP 🤔 

Can you access: https://api.ipify.org ?

**Comment by nmvh5** [comment]
Yes, that does work.

**Comment by nickbabcock** [comment]
Perfect, I'll need to put in some work to allow one to switch providers.

**Comment by nmvh5** [comment]
That'd be great. I've been using your app for quite a while and this is the first issue I've had. Thanks for the quick followup!

**Comment by nickbabcock** [comment]
Alright, I released 0.3.0. You should be up and working if you prepend the following to the dness.conf

```toml
ip_resolver = "ipify"
```

Let me know if you have any more issues

**Comment by lukyjay** [comment]
I encountered this issue but it was my firewall blocking port 53. Leaving a note for future users who find this issue while troubleshooting, ensure you have a 'pass' rule for port 53 to your Dness IP
